daffodil-cli/src/it/scala/org/apache/daffodil/tunables/TestCLITunables2.scala

```diff
@@ -80,4 +82,52 @@ class TestCLITunables2 {
       shell.close()
     }
   }
+
+  @Test def test_CLI_Parsing_ReloadingDoesNotRepeatWarnings(): Unit = {
+    val schemaFile = Util.daffodilPath("daffodil-cli/src/it/resources/org/apache/daffodil/CLI/suppressWarnTest.dfdl.xsd")
+    val testSchemaFile = if (Util.isWindows) Util.cmdConvert(schemaFile) else schemaFile
+    val compiledProcFile = Util.newTempFile("savedProc", ".bin")
+    compiledProcFile.deleteOnExit()
+    val compiledProcFilePath = compiledProcFile.getAbsolutePath()
+    //
+    do {
+      val shell = Util.start("")
+      try {
+        // note: 2>&1 is shell-speak for "connect stderr into stdout"
+        val cmd = String.format("""%s save-parser -s %s %s 2>&1""", Util.binPath, testSchemaFile, compiledProcFilePath)
+        shell.sendLine(cmd)
+        //
+        // Saving the processor should compile and issue SDWs which we should see
+        // in the expected output
+        //
+        shell.expect(contains("""Schema Definition Warning"""))
+        Util.expectExitCode(ExitCode.Success, shell)
+        shell.sendLine("exit")
+        shell.expect(eof)
+      } finally {
+        shell.close()
+      }
+    } while (false) // workaround scala local block bug.
+    do {
+      val shell = Util.start("")
+      try {
+        // note: 2>&1 is shell-speak for "connect stderr into stdout"
+        val cmd = String.format("""echo a,b| %s parse -P %s 2>&1""", Util.binPath, compiledProcFilePath)
+        shell.sendLine(cmd)
+        shell.sendLine("exit")
+        val output = shell.expect(eof).getBefore
```

Comment:
This is nice! A good technique to know to just get all output to a string. It's kindof the only good way to check that output doesn't exist.